commit 64cfdbc4e18161aa86b930fb83ad0b749703c9d3
Author: Or Gerlitz <ogerlitz@mellanox.com>
Date:   Wed Aug 8 15:48:07 2018 -0700

    net/mlx5e: Properly check if hairpin is possible between two functions
    
    [ Upstream commit 816f670623692b5da2787f278cbfdb331ed29b8a ]
    
    The current check relies on function BDF addresses and can get
    us wrong e.g when two VFs are assigned into a VM and the PCI
    v-address is set by the hypervisor.
    
    Fixes: 5c65c564c962 ('net/mlx5e: Support offloading TC NIC hairpin flows')
    Signed-off-by: Or Gerlitz <ogerlitz@mellanox.com>
    Reported-by: Alaa Hleihel <alaa@mellanox.com>
    Tested-by: Alaa Hleihel <alaa@mellanox.com>
    Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c b/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c
index a0ba6cfc9092..290fc6f9afc1 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c
@@ -1907,15 +1907,15 @@ static bool actions_match_supported(struct mlx5e_priv *priv,
 static bool same_hw_devs(struct mlx5e_priv *priv, struct mlx5e_priv *peer_priv)
 {
 	struct mlx5_core_dev *fmdev, *pmdev;
-	u16 func_id, peer_id;
+	u64 fsystem_guid, psystem_guid;
 
 	fmdev = priv->mdev;
 	pmdev = peer_priv->mdev;
 
-	func_id = (u16)((fmdev->pdev->bus->number << 8) | PCI_SLOT(fmdev->pdev->devfn));
-	peer_id = (u16)((pmdev->pdev->bus->number << 8) | PCI_SLOT(pmdev->pdev->devfn));
+	mlx5_query_nic_vport_system_image_guid(fmdev, &fsystem_guid);
+	mlx5_query_nic_vport_system_image_guid(pmdev, &psystem_guid);
 
-	return (func_id == peer_id);
+	return (fsystem_guid == psystem_guid);
 }
 
 static int parse_tc_nic_actions(struct mlx5e_priv *priv, struct tcf_exts *exts,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1013_64cfdbc4e18161aa86b930fb83ad0b749703c9d3.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1013_64cfdbc4e18161aa86b930fb83ad0b749703c9d3.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
